packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -275,6 +294,62 @@ export class AmplifyGraphqlApi extends Construct {
     outputStorageStrategy.addBackendOutputEntry(graphqlOutputKey, output);
   }
 
+  /**
+   * Gets the deployment information from the context.
+   * @returns the deployment identifier.
+   */
+  private getDeploymentIdentifier(): DeploymentIdentifier {
+    const deploymentType = this.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE);
+    const expectedDeploymentTypes = [this.SANDBOX_DEPLOYMENT_TYPE, this.BRANCH_DEPLOYMENT_TYPE];
+
+    // Throw an error if the deployment type is neither sandbox nor branch.
+    if (!expectedDeploymentTypes.includes(deploymentType)) {
```

Comment:
If the GQL construct is used outside of the context of an Amplify backend (ie as a construct in a native CDK project), then these context props will not be present

Reply:
To clarify, you are referring to the class variables `this.SANDBOX_DEPLOYMENT_TYPE, this.BRANCH_DEPLOYMENT_TYPE` right?

Reply:
Correct, none of these context values would be present in a native CDK project. The construct should probably treat absence of these values the same as a "branch" deployment

Reply:
Let me rephrase what you are suggesting here. Are you saying that in the case of a native CDK project, where `this.SANDBOX_DEPLOYMENT_TYPE, this.BRANCH_DEPLOYMENT_TYPE` are absent, `getDeploymentIdentifier()` should treat this case as a branch deployment and return a `BranchDeploymentIdentifier`?

I see another workaround for the absence of `this.SANDBOX_DEPLOYMENT_TYPE, this.BRANCH_DEPLOYMENT_TYPE` is to make them local inside `getDeploymentIdentifier()`. Will this solve the problem?

Reply:
You should only emit the warning if a deployment type is actually present:

```
if (deploymentType && !expectedDeploymentTypes.includes(deploymentType)) {
```

Reply:
As far as I recall data construct was supposed to be usable outside of amplify in raw CDK apps.

Should this logic be 
```
if (deploymentType) {
  if(deploymentType is invalid) {
   // If we're in Amplify Gen2 app, customer doesn't control these keys, fail fast.
   throw Error();
  }
} else {
   // you are in raw CDK app, none of Amplify CDK context keys is present, which is fine, default to something reasonable.
}
```